src/expression/expression.ts

```diff
@@ -18,7 +18,7 @@ export interface Expression {
 
 export type ExpressionParser = (args: ReadonlyArray<unknown>, context: ParsingContext) => Expression;
 export type ExpressionRegistration = {
-    new (...args: any): Expression;
+    prototype: Expression;
```

Comment:
tweaked this because my `Semiliteral` class is `abstract` and therefore doesn't have a constructor